filebeat/input/log/input.go

```diff
@@ -87,11 +88,21 @@ func NewInput(
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
+		if cleanupNeeded {
+			out.Close()
+		}
+	}()
 
 	// stateOut will only be unblocked if the beat is shut down.
 	// otherwise it can block on a full publisher pipeline, so state updates
 	// can be forwarded correctly to the registrar.
 	stateOut := channel.CloseOnSignal(channel.SubOutlet(out), context.BeatDone)
+	defer func() {
+		if cleanupNeeded {
+			stateOut.Close()
+		}
+	}()
```

Comment:
These outlets should be probably created on `Run()`, but they are needed now to check the configuration. I'd fix it like this by now, and would do further refactors on future PRs.